src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -410,5 +415,19 @@ public async Task<bool> HasPropertiesAsync(Uri blobUri, string contentType, stri
 
             return false;
         }
+
+        private CloudBlockBlob GetBlockBlobReference(string blobName)
+        {
+            var blob = _directory.GetBlockBlobReference(blobName);
+
+            ApplyBlobRequestOptions(blob, _directory);
+
+            return blob;
+        }
+
+        private static void ApplyBlobRequestOptions(CloudBlockBlob blob, CloudBlobDirectory directory)
```

Comment:
You could simplify this to:

    private void ApplyBlobRequestOptions(CloudBlockBlob blob)

...and just use `_directory` in the implementation, similar to the method above.

Reply:
Good point. Fixed.